libs/design/src/core/openable/openable.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * An interface for giving a component the ability to display an open UI.
+ * In order to be openable, the class must implement this property.
+ */
+export interface DaffOpenable {
+  /** Whether the component is open or not */
+  open: boolean;
+
+  /** Sets reveal on the component */
+  reveal: () => void;
+
+  /** Sets hide on the component */
+  hide: () => void;
+
+  /** Sets toggle on the component */
+  toggle: () => void;
+}
```

Comment:
These docs aren't helpful. Especially for `toggle`, they should say something about the expected behavior of the method.

Reply:
updated